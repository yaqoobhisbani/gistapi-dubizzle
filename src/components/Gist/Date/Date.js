import React from "react";
import styled from "styled-components";
import { formatDate } from "../../../utils";

const Date = ({ label, date, style }) => {
  return (
    <Wrapper style={style}>
      <span>{label}</span>
      <span>{formatDate(date)}</span>
    </Wrapper>
  );
};

const Wrapper = styled.p`
  color: ${(props) => props.theme.gray};
  font-size: 14px;
`;

export default Date;
