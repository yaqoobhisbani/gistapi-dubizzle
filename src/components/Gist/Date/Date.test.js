import React from "react";
import { render } from "@testing-library/react";
import Date from "./Date";

describe("Date", () => {
  test("renders label and formatted date", () => {
    const mockLabel = "Date:";
    const mockDate = "2023-03-30T20:51:53Z";
    const { getByText } = render(<Date label={mockLabel} date={mockDate} />);
    const labelElement = getByText(mockLabel);
    const dateElement = getByText("2/31/2023");
    expect(labelElement).toBeInTheDocument();
    expect(dateElement).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const mockLabel = "Date:";
    const mockDate = "2023-03-30T20:51:53Z";
    const { container } = render(<Date label={mockLabel} date={mockDate} />);
    expect(container).toMatchSnapshot();
  });
});
