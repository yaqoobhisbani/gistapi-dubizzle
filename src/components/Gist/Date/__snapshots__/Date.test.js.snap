// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date matches snapshot 1`] = `
<div>
  <p
    class="sc-beqWaB yDHQF"
  >
    <span>
      Date:
    </span>
    <span>
      2/31/2023
    </span>
  </p>
</div>
`;
